From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Dreeam <61569423+Dreeam-qwq@users.noreply.github.com>
Date: Thu, 15 Jan 2026 18:07:51 -0500
Subject: [PATCH] Fix paper tests

snakeyaml deprecated the external util class `Base64Coder` since snakeyaml 2.3
See: https://github.com/snakeyaml/snakeyaml/commit/3f0bedad5917490a1f7ef56a0920b5b93b214ade

It replaced it using Java's built-in Base64 util (java.util.Base64), which is better and more standard.

In here, I use Base64.getMimeDecoder() for the base64 decoding, to be compatible with old data that is using the old
standard (the Base64Coder is using MIME type).
See: https://docs.oracle.com/javase/8/docs/api/java/util/Base64.html#:~:text=MIME

And use the basic Base64 encoder to encode the data.

diff --git a/src/test/java/org/bukkit/craftbukkit/inventory/ItemStackTest.java b/src/test/java/org/bukkit/craftbukkit/inventory/ItemStackTest.java
index 1d4d573a782aed2a2668d7b98ed8973e8d7aa422..002b249103760227a57b376ed1ecfb298e66fe20 100644
--- a/src/test/java/org/bukkit/craftbukkit/inventory/ItemStackTest.java
+++ b/src/test/java/org/bukkit/craftbukkit/inventory/ItemStackTest.java
@@ -27,7 +27,6 @@ import org.bukkit.util.io.BukkitObjectOutputStream;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.Arguments;
 import org.junit.jupiter.params.provider.MethodSource;
-import org.yaml.snakeyaml.external.biz.base64Coder.Base64Coder;
 
 @AllFeatures
 public class ItemStackTest {
@@ -510,7 +509,7 @@ public class ItemStackTest {
             }
         }
 
-        final String data = new String(Base64Coder.encode(out.toByteArray()));
+        final String data = java.util.Base64.getEncoder().encodeToString(out.toByteArray()); // Leaf - Fix paper tests
 
         ObjectInputStream ois = null;
 
